packages/shorebird_cli/lib/src/validators/shorebird_flutter_validator.dart

```diff
@@ -14,38 +15,57 @@ class ShorebirdFlutterValidator extends DoctorValidator {
 
   @override
   Future<List<ValidationIssue>> validate() async {
+    final issues = <ValidationIssue>[];
+
     if (!ShorebirdPaths.flutterDirectory.existsSync()) {
       final message =
           'No Flutter directory found at ${ShorebirdPaths.flutterDirectory}';
-      return [
+      issues.add(
         ValidationIssue(
           severity: ValidationIssueSeverity.error,
           message: message,
         ),
-      ];
+      );
     }
 
     if (!await _flutterDirectoryIsClean()) {
-      return [
+      issues.add(
         ValidationIssue(
           severity: ValidationIssueSeverity.warning,
           message: '${ShorebirdPaths.flutterDirectory} has local modifications',
         ),
-      ];
+      );
     }
 
     if (!await _flutterDirectoryTracksStable()) {
       final message =
           '${ShorebirdPaths.flutterDirectory} is not on the "stable" branch';
-      return [
+      issues.add(
+        ValidationIssue(
+          severity: ValidationIssueSeverity.warning,
+          message: message,
+        ),
+      );
+    }
+
+    final shorebirdFlutterVersion = await _shorebirdFlutterVersion();
+    final pathFlutterVersion = await _pathFlutterVersion();
+
+    if (shorebirdFlutterVersion != pathFlutterVersion) {
+      final message = '''
+Shorebird Flutter and the Flutter on your path are different versions.
+\tShorebird Flutter: $shorebirdFlutterVersion
+\tSystem Flutter:    $pathFlutterVersion''';
+
```

Comment:
"This can cause unexpected behavior if the version gap is wide.  If you're seeing this unexpectedly, please let us know on Shorebird discord!"

Reply:
Done.